### PR TITLE
use bulk /txs endpoint to check cached rbf tx status

### PR DESCRIPTION
**Pull request by mononaut**
_(builds on #4158)_

After we load RBF data in from the cache, we need to check if anything got mined or removed from the mempool while we were offline.

Currently, we fetch these transactions one by one, but that can be slow when there are a lot of transactions in the RBF cache.

This PR uses a new esplora `/internal-api/txs` endpoint to fetch the transactions in bulk, which speeds up the process significantly.

(we can't use the `/internal-api/mempool/txs` esplora endpoint because we need differentiate between transactions being confirmed and evicted from the mempool)

### CODE REVIEW

File: backend/src/api/bitcoin/esplora-api.ts [hunk 1]
```diff
@@ -88,6 +88,10 @@ class ElectrsApi implements AbstractBitcoinApi {
     return this.$queryWrapper<IEsploraApi.Transaction>(config.ESPLORA.REST_API_URL + '/tx/' + txId);
   }
 
+  async $getRawTransactions(txids: string[]): Promise<IEsploraApi.Transaction[]> {
+    return this.$postWrapper<IEsploraApi.Transaction[]>(config.ESPLORA.REST_API_URL + '/txs', txids, 'json');
```
Comment (softsimon) [inline backend/src/api/bitcoin/esplora-api.ts:92]: Endpoint doesn't exist ,, `/mempool` is missing?

**Review by softsimon — CHANGES_REQUESTED** [review]

**Comment by softsimon** [comment]
Requires https://github.com/mempool/electrs/pull/33

**Comment by mononaut** [comment]
re-rebased on #4158

File: backend/src/api/rbf-cache.ts [hunk 2]
```diff
@@ -480,14 +480,16 @@ class RbfCache {
     };
 
     if (config.MEMPOOL.BACKEND === 'esplora') {
-      const sliceLength = 10000;
+      const sliceLength = 1000;
```
Comment (wiz) [inline backend/src/api/rbf-cache.ts:483]:
```suggestion
      const sliceLength = 250;
```

**Review by wiz — APPROVED** [review]
Tested ACK with requested changes @ 5998b54

Seems to mostly work with 1000 chunk size, but still fails occasionally so let's merge using 250 chunk size and see how it goes

```
Nov 12 10:03:18 [51122] DEBUG: RUST makeBlockTemplates completed in 13.296 seconds
Nov 12 10:03:26 [51122] INFO: Restoring rbf data from disk cache
Nov 12 10:03:28 [51122] INFO: Starting statistics service
Nov 12 10:03:28 [51122] DEBUG: [Mining] Inserted 0 MtGox USD weekly price history into db
Nov 12 10:03:29 [51122] DEBUG: [Mining] Fetching daily price history from exchanges and saving missing ones into the database
Nov 12 10:03:29 [51122] DEBUG: Updated orphaned blocks cache. Found 6 orphaned blocks
Nov 12 10:03:29 [51122] NOTICE: Mempool Server is running on port 8999
Nov 12 10:03:30 [51122] DEBUG: [Mining] Inserted 0 daily historical prices into the db
Nov 12 10:03:30 [51122] DEBUG: [Mining] Fetching hourly price history from exchanges and saving missing ones into the database
Nov 12 10:03:31 [51122] DEBUG: [Mining] Inserted 0 hourly historical prices into the db
Nov 12 10:03:32 [51122] DEBUG: Memory usage: 4.40 GB / 16.05 GB
Nov 12 10:03:38 [51122] WARN: esplora request failed undefined /bitcoin/socket/esplora-bitcoin-mainnet/internal/txs
Nov 12 10:03:38 [51122] WARN: timeout of 10000ms exceeded
Nov 12 10:03:38 [51122] ERR: failed to fetch or process 1000 cached rbf transactions
Nov 12 10:03:39 [51122] WARN: esplora request failed undefined /bitcoin/socket/esplora-bitcoin-mainnet/mempool/txids
Nov 12 10:03:39 [51122] WARN: timeout of 10000ms exceeded
Nov 12 10:03:39 [51122] DEBUG: Exception in runMainUpdateLoop() (count: 1). Retrying in 1 sec. Reason: timeout of 10000ms exceeded. Stack trace: AxiosError: timeout of 10000ms exceeded
    at RedirectableRequest.handleRequestTimeout (/mempool/mainnet/backend/node_modules/axios/dist/node/axios.cjs:3120:16)
    at RedirectableRequest.emit (node:events:514:28)
    at Timeout.<anonymous> (/mempool/mainnet/backend/node_modules/follow-redirects/index.js:169:12)
    at listOnTimeout (node:internal/timers:573:17)
    at process.processTimers (node:internal/timers:514:7)
Nov 12 10:03:39 [51122] DEBUG: AxiosError: timeout of 10000ms exceeded
Nov 12 10:03:48 [51122] DEBUG: Initial difficulty adjustment data set.
Nov 12 10:03:48 [51122] DEBUG: New block found (#816428)!
Nov 12 10:03:48 [51122] WARN: esplora request failed undefined /bitcoin/socket/esplora-bitcoin-mainnet/internal/txs
Nov 12 10:03:48 [51122] WARN: timeout of 10000ms exceeded
Nov 12 10:03:48 [51122] ERR: failed to fetch or process 1000 cached rbf transactions
Nov 12 10:03:49 [51122] DEBUG: Updated orphaned blocks cache. Found 6 orphaned blocks
Nov 12 10:03:53 [51122] DEBUG: 907 of 4832 found in mempool. 3925 fetched through backend service.
Nov 12 10:03:53 [51122] DEBUG: 0 websocket clients | 0 connected | 0 disconnected | (+0)
Nov 12 10:03:55 [51122] DEBUG: RUST updateBlockTemplates completed in 2.03 seconds
Nov 12 10:03:55 [51122] DEBUG: New block found (#816429)!
Nov 12 10:03:56 [51122] DEBUG: Updated orphaned blocks cache. Found 6 orphaned blocks
Nov 12 10:03:56 [51122] DEBUG: fetched 1000 cached rbf transactions
Nov 12 10:03:56 [51122] DEBUG: processed 1000 cached rbf transactions
Nov 12 10:03:56 [51122] DEBUG: fetched 1000 cached rbf transactions
Nov 12 10:03:56 [51122] DEBUG: processed 1000 cached rbf transactions
Nov 12 10:04:04 [51122] DEBUG: fetched 1000 cached rbf transactions
Nov 12 10:04:04 [51122] DEBUG: processed 1000 cached rbf transactions
Nov 12 10:04:10 [51122] DEBUG: 3271 of 5121 found in mempool. 1850 fetched through backend service.
Nov 12 10:04:11 [51122] DEBUG: fetched 1000 cached rbf transactions
Nov 12 10:04:11 [51122] DEBUG: processed 1000 cached rbf transactions
Nov 12 10:04:11 [51122] DEBUG: 0 websocket clients | 0 connected | 0 disconnected | (+0)
Nov 12 10:04:13 [51122] DEBUG: RUST updateBlockTemplates completed in 1.826 seconds
```

**Pull request merged**